active/TestInsecureHTTPVerbs.py

```diff
@@ -0,0 +1,71 @@
+http://crackme.cenzic.com
+
+and other apps from https://www.owasp.org/index.php/OWASP_Vulnerable_Web_Applications_Directory_Project#tab=On-Line_apps
+
+"""
+
+insecureverbs=["HEAD", "OPTIONS", "PUT", "DELETE", "TRACE", "CONNECT", "DEBUG", "MOVE", "SEARCH", "PATCH", "MKCOL", "COPY", "LOCK", "UNLOCK", "ARBIT", "XXXX", "12AB"]
+acceptedhttpstatuscodesforinsecureverbs=[301, 302, 404, 405, 403, 400, 501] # if we get a response code that is not a part of this array, then flag as potentially vulnerable
+activevulnerabilitytitle='Potentially Insecure HTTP Verb allowed'
+activevulnerabilityfulldescription='Some of the HTTP methods can potentially pose a security risk for a web application, as they allow an attacker to modify the files stored on the web server and, in some scenarios, steal the credentials of legitimate users.' + 'Insecure configuration can possibly lead to web server compromise and website defacement. ' + 'If an application needs one or more of the potentially insecure HTTP methods, such as for REST Web Services (which may require PUT or DELETE), it is important to check that their usage is properly limited to trusted users and safe conditions. ' + 'http://security.stackexchange.com/questions/21413/how-to-exploit-http-methods.'
+activevulnerabilitysolution = 'Configure the web server to allow insecure methods like DELETE and PUT only for the relevant resources. ' + 'If your application does not need HTTP methods other than GET and POST, consider disabling the unused HTTP methods.'
+printdebugmessages = True
+#printdebugmessages= False
+
+def PrintAlerts(sas, msg, uri, insecureverb, responsestatuscode, responsestatusmessage):
+	attackevidence = 'VERB: \t' + insecureverb + '\t-- HTTP STATUS CODE: '+ str(responsestatuscode) +' -- \tMESSAGE: ' + responsestatusmessage  + '\t\t-- POTENTIALLY VULNERABLE: YES\n'
+	if printdebugmessages:
+		print (attackevidence)
+	sas.raiseAlert(1, 2, activevulnerabilitytitle, activevulnerabilityfulldescription, uri, 'HTTP VERB', 'ZAP sent an HTTP request with method - ' + insecureverb, '', activevulnerabilitysolution, attackevidence, 0, 0, msg);
+
+	
+def ProcessAndPrintAlert(sas, msg, uri, insecureverb):
+	if not msg.getResponseHeader().getStatusCode() in acceptedhttpstatuscodesforinsecureverbs:
+		PrintAlerts(sas, msg, uri, insecureverb, msg.getResponseHeader().getStatusCode(), msg.getResponseHeader().getReasonPhrase())
+
+
+def PrepareHttpRequest(msg, insecureverb):
+	msg.mutateHttpMethod(insecureverb)
+	if insecureverb == "POST" or insecureverb == "PUT":
+		msg.setRequestBody("bodytext")
+	if printdebugmessages:
+		print('After mutating resulting HTTP VERB: -- \t' + msg.getRequestHeader().getMethod() + '\n')
+
+	
+def TestTheURIForInsecureVerbs(sas, msg, uri, insecureverbs):
+	for insecureverb in insecureverbs:
+		try:
+			if printdebugmessages:
+				print ('Testing Initiated for the HTTP VERB: -- \t' + insecureverb + '\n')
+			PrepareHttpRequest(msg, insecureverb)
+			sas.sendAndReceive(msg)
```

Comment:
This should call `sendAndReceive(msg, False)`, to not follow redirects. Sorry, for noticing in the first review.